### PR TITLE
Fixed global event override which caused issues with other addons

### PR DESCRIPTION
**Pull request by Koviko**
Should fix issue with FishingBuddy and Fishing Ace mentioned in #219.

### CODE REVIEW

**Comment by brittyazel** [comment]
Awesome I'll review this once I'm home from work. Can you quickly explain to me what the logic behind this change is just so I know? What was causing the issue in the first place and how does this fix it?

**Comment by Koviko** [comment]
Sure thing. But bare with me if I misspeak, as I am not very familiar with Lua or with WoW scripting.

LibFishing-1.0, the library responsible for casting Fishing with a double-right-click, needed to hook into `WorldFrame:OnMouseDown()`. However, Neuron seemed to be replacing `WorldFrame:OnMouseDown()`'s event handler which prevented any `WorldFrame:OnMouseDown()` hooks from firing.

The replacement occurred in `ACTIONBUTTON` and replaced `WorldFrame:OnMouseDown()`'s event handler with `ACTIONBUTTON:OnMouseDown()`. This seemed doubly-wrong as, not only was the event handler being replaced, it was being re-replaced with every generated `ACTIONBUTTON`. Also, `ACTIONBUTTON:OnMouseDown()` never required a reference to the calling `ACTIONBUTTON`.

With those issues in mind, I:

1. Moved `ACTIONBUTTON:OnMouseDown()` to `Neuron:OnMouseDown()`, as it's only purpose was clearing the `Neuron.macroDrag` table,
1. Moved the `WorldFrame:OnMouseDown()` change out of `ACTIONBUTTON` and over to `Neuron`, and
1. Hooked into the `WorldFrame:OnMouseDown()` event instead of replacing it's event handler.

I also left `ACTIONBUTTON:OnMouseDown()` in place just in case the handler needs to be used in the future. A good pull request changes as little as is necessary. <3

**tl;dr** I changed `WorldFrame:SetScript(...)` to `Neuron:HookScript(WorldFrame, ...)` and moved `OnMouseDown()` code from `ACTIONBUTTON` to `Neuron`.

**Comment by Koviko** [comment]
Also, I just pushed a fix for where I typed "NnMouseDown" instead of "OnMouseDown" in a comment. lol

I'm hoping the push request properly includes this.

**Comment by brittyazel** [comment]
Excellent! I made a few changes, I hope you don't mind. I wanted to keep the OnMouseClick in the ACTIONBUTTON class just for organization. I also added a safety check to not re-hook if we're already hooked to that function.

So it seems like the "fix" was in using HookScript instead of SetScript which makes a lot of sense :-) That was an oversight on my part from way back when I first joined the project lol

**Comment by brittyazel** [comment]
Actually hooking onto WorldFrame the way I did was effective but kinda hacky. In the future I want to make this better in general, this is a great solution for the time being though :-) Can you verify that it still works with your fishing addons?

**Comment by Koviko** [comment]
That makes sense. `OnMouseDown` did feel very out-of-place inside of Neuron.lua. lol

And yes, the changes still work with my fishing addon!

**Comment by brittyazel** [comment]
hah awesome. Well if you ever get the bug again, there's lots to do :-) I appreciate you taking the time to sort out this bug
